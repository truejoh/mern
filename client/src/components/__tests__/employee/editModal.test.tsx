import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import EditModal from '../../employee/modals/editModal';

describe('Employee Edit Modal', () => {
  it('renders component', () => {
    const props = {
      visible: true,
      item: {
        _id: '123',
        name: 'Employee A',
        number: 1234567,
        accountHolder: 'Holder A',
        accountType: 'Checking',
        accountNumber: 7654321,
      },
      onOk: () => {},
      onCancel: () => {},
    };
    const component = shallow(<EditModal {...props} />);

    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders props', () => {
    const props = {
      visible: true,
      item: {
        _id: '123',
        name: 'Employee A',
        number: 1234567,
        accountHolder: 'Holder A',
        accountType: 'Checking',
        accountNumber: 7654321,
      },
      onOk: () => {},
      onCancel: () => {},
    };
    const component = shallow(<EditModal {...props} />);

    expect(component.prop('visible')).toEqual(true);
    expect(component.prop('onOk')).toBeInstanceOf(Function);
    expect(component.prop('onCancel')).toBeInstanceOf(Function);
  });
});
