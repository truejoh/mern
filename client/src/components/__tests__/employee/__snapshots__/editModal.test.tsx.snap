// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee Edit Modal renders component 1`] = `
<Modal
  confirmLoading={false}
  maskTransitionName="fade"
  okType="primary"
  onCancel={[Function]}
  onOk={[Function]}
  title={
    <InjectIntl(InjectMassage)
      id="employee.modal.edit.title"
    />
  }
  transitionName="zoom"
  visible={true}
  width={520}
>
  <Form
    colon={true}
    hideRequiredMark={false}
    layout="horizontal"
    onSubmit={[Function]}
  >
    <FormItem
      className="form-element"
      hasFeedback={false}
      label={
        <InjectIntl(InjectMassage)
          id="employee.modal.step.employee.name"
        />
      }
    >
      <Input
        defaultValue="Employee A"
        disabled={false}
        onChange={[Function]}
        placeholder="input name"
        type="text"
      />
    </FormItem>
    <FormItem
      className="form-element"
      hasFeedback={false}
      label={
        <InjectIntl(InjectMassage)
          id="employee.modal.step.employee.number"
        />
      }
    >
      <InputNumber
        defaultValue={1234567}
        onChange={[Function]}
        placeholder="input number"
        step={1}
      />
    </FormItem>
    <FormItem
      className="form-element"
      hasFeedback={false}
      label={
        <InjectIntl(InjectMassage)
          id="employee.modal.step.account.holder"
        />
      }
    >
      <Input
        defaultValue="Holder A"
        disabled={false}
        onChange={[Function]}
        placeholder="input account holder"
        type="text"
      />
    </FormItem>
    <FormItem
      className="form-element"
      hasFeedback={false}
      label={
        <InjectIntl(InjectMassage)
          id="employee.modal.step.account.type"
        />
      }
    >
      <Select
        choiceTransitionName="zoom"
        defaultValue="Checking"
        onChange={[Function]}
        showSearch={false}
        transitionName="slide-up"
      >
        <Option
          value="Saving"
        >
          <InjectIntl(InjectMassage)
            id="employee.modal.step.account.type.saving"
          />
        </Option>
        <Option
          value="Checking"
        >
          <InjectIntl(InjectMassage)
            id="employee.modal.step.account.type.checking"
          />
        </Option>
      </Select>
    </FormItem>
    <FormItem
      className="form-element"
      hasFeedback={false}
      label={
        <InjectIntl(InjectMassage)
          id="employee.modal.step.account.number"
        />
      }
    >
      <InputNumber
        defaultValue={7654321}
        onChange={[Function]}
        placeholder="input account number"
        step={1}
      />
    </FormItem>
  </Form>
</Modal>
`;
